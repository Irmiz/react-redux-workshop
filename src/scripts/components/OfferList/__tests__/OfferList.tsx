import * as React from 'react';
import OfferList from 'components/OfferList';
import renderer = require('react-test-renderer');
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';

const mockStore = configureStore();

describe('Offers component', () => {
    it('renders a list', () => {
        const store = mockStore({
            offers: [
                {
                    name: 'Bob'
                },
                {
                    name: 'Alice'
                }
            ]
        });
        const offersComponent = renderer.create(
            <Provider store={store}>
                <OfferList />
            </Provider>
        ).toJSON();

        expect(offersComponent).toMatchSnapshot();
    });

    it('renders empty list', () => {
        const store = mockStore({
            offers: []
        });
        const offersComponent = renderer.create(
            <Provider store={store}>
                <OfferList />
            </Provider>
        ).toJSON();

        expect(offersComponent).toMatchSnapshot();
    });
});
