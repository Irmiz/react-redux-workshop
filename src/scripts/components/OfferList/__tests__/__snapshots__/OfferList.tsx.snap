exports[`Offers component renders a list 1`] = `
<div>
  <ul>
    <li>
      Bob
    </li>
    <li>
      Alice
    </li>
  </ul>
</div>
`;

exports[`Offers component renders empty list 1`] = `
<div>
  <ul />
</div>
`;
